core/chat/events/events.go

```diff
@@ -12,51 +12,72 @@ import (
 	"github.com/yuin/goldmark/extension"
 	"github.com/yuin/goldmark/renderer/html"
 	"mvdan.cc/xurls"
+
+	"github.com/owncast/owncast/core/user"
+	log "github.com/sirupsen/logrus"
 )
 
-// ChatEvent represents a single chat message.
-type ChatEvent struct {
-	ClientID string `json:"-"`
-
-	Author      string    `json:"author,omitempty"`
-	Body        string    `json:"body,omitempty"`
-	RawBody     string    `json:"-"`
-	ID          string    `json:"id"`
-	MessageType EventType `json:"type"`
-	Visible     bool      `json:"visible"`
-	Timestamp   time.Time `json:"timestamp,omitempty"`
-	Ephemeral   bool      `json:"ephemeral,omitempty"`
+// EventPayload is a generic key/value map for sending out to chat clients.
+type EventPayload map[string]interface{}
+
+type OutboundEvent interface {
+	GetBroadcastPayload() EventPayload
+	GetMessageType() EventType
+}
+
+// Event is any kind of event.  A type is required to be specified.
+type Event struct {
+	Type      string    `json:"type"`
```

Comment:
Can we pin this to be of type `EventType`?

```suggestion
	Type      EventType `json:"type"`
```